core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/TimeoutExceptionHandlingStage.java

```diff
@@ -118,6 +119,10 @@ private Exception handleTimeoutCausedException(RequestExecutionContext context,
             return AbortedException.create("Thread was interrupted", e);
         }
 
+        if (e instanceof UncheckedIOException) {
+            return new IOException(e.getCause());
+        }
+
```

Comment:
Do we need this logic? If the stream is closed by the timer, then it will return from line 107 or 118, right?